### ISSUE TITLE
bug with painting mentions over PDF?

### ISSUE THREAD

**Issue by jameshowison** [softcite/softcite_kb#11]
I was playing around and noticed this issue, seems like the mentions are being painted on the wrong page, perhaps?  or maybe the underlying PDF changed?

https://cloud.science-miner.com/software_kb/frontend/document.html?id=611a11d8e8d4855847028e44

Hopefully this screenshot shows the issue (This is with Firefox 109.0.1 on Mac OS btw)

![Screen Shot 2023-04-05 at 12 13 07 PM](https://user-images.githubusercontent.com/91986/230154734-dda17c61-4b1b-4c6c-be6e-73a325948b61.png)

**Comment by kermitt2** [comment]
Indeed, in the current version it can happen when the underlying paper has changed. The GUI fetches the Open Access PDF at URL given by a call to the Unpaywall API, but if the version of the PDF has changed between the time the annotations was produced and stored in the KB and now, we have position "mismatch" for the annotations. 

For example, the annotations produced in 2021 was made with the preprint, and now Unpaywall points to the Gold version of the article. 

I store a hash for the PDF originally annotated, so I could detect normally if the fetched PDF is still the same of not - but this is not yet implemented. Then if the PDF mismatch is detected we could imagine a fallback solution, like trying alternative URL to get other version of the PDF or re-annotating on the fly.

**Comment by jameshowison** [comment]
Makes sense.  Maybe we should store something that enables us to hit the same version (if at all possible)? That's hard without storing the actual article (which we can't do), but perhaps re-trying the URL (or following the metadata to the same version).

But also highlight that unpaywall has a new version?
